hw/mcu/dialog/da1469x/src/hal_system_start.c

```diff
@@ -61,3 +70,91 @@ hal_system_restart(void *img_start)
 
     hal_system_start(img_start);
 }
+
+#if MYNEWT_VAL(BOOT_CUSTOM_START) && MCUBOOT_MYNEWT
+#define IMAGE_TLV_AES_NONCE   0x50
+#define IMAGE_TLV_SECRET_ID   0x60
+
+sec_text_ram_core void
+boot_custom_start(uintptr_t flash_base, struct boot_rsp *rsp)
+{
+    int rc;
+    struct image_tlv_iter it;
+    const struct flash_area *fap;
+    uint32_t off;
+    uint16_t len;
+    uint8_t type;
+    uint8_t buf[16];
+    uint8_t key;
+    uint32_t nonce[2];
+    bool has_aes_nonce;
+    bool has_secret_id;
+    DMA_Type *dma_regs = DMA;
+    uint32_t  jump_addr = flash_base + rsp->br_image_off +
```

Comment:
yeah i can collapse this down.  I didn't simplify this down when I moved it around.

Reply:
now i remember why, we are disabling the decrypt on the fly for mcuboot at the time we are doing this.  And we have to have everything we need on the stack, so i was paranoid to avoid any reference to anything in flash.